examples/polyglot-zipkin/java-dropwizard/src/main/java/org/hawkular/apm/example/dropwizard/rest/AsyncHandler.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.apm.example.dropwizard.rest;
+
+import java.util.Objects;
+import java.util.Random;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.container.AsyncResponse;
+import javax.ws.rs.container.Suspended;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+
+import com.github.kristofa.brave.Brave;
+import com.github.kristofa.brave.BraveExecutorService;
+
+/**
+ * @author Pavol Loffay
+ */
+@Path("/")
+public class AsyncHandler {
+
+    private final Client client;
+    private final BraveExecutorService braveExecutorService;
+
+    private ThreadLocal<Random> randomThreadLocal = new ThreadLocal<Random>() {
+        @Override
+        protected Random initialValue() {
+            return new Random();
+        }
+    };
+
+
+    public AsyncHandler(Client client, Brave brave) {
+        this.client = Objects.requireNonNull(client);
+        this.braveExecutorService = new BraveExecutorService(Executors.newFixedThreadPool(10),
+                brave.serverSpanThreadBinder());
+    }
+
+    @GET
+    @Path("/async")
+    public Response asyncRoot(@Context HttpServletRequest request) {
+
+        final int serverPort = request.getServerPort();
+
+        braveExecutorService.execute(() -> {
+            try {
+                TimeUnit.MILLISECONDS.sleep(1000);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+            client.target("http://127.0.0.1:" + serverPort + "/asyncEnd").request().get();
```

Comment:
Minor - I was initially confused by the path names - End and Middle, and they don't tie up with the method names. Might be better to go back to the letters, but possibly "asyncServiceA" and "asyncServiceB" - so then no confusion with what 'end' and 'middle' means?